Changed form,dashboard

rav-indra opened the pull request:
- Changed form style
-  Added new page to show sold properties
- Changed dashboard style
- Revoke edit and delete access from seller after property sold

santucorephp committed:
Changed form,dashboard

ravin-essence commented on app/views/dashboard/_buyer.html.erb:
```diff
@@ -1,15 +1,45 @@
-<h1 style="margin-top: 120px; text-align:center;">Purchased Properties</h1>
 <% props = Property.where(buyer_id: current_user.id) %>
```
View is only for presenting the data, Not for fetching or collecting the data. So please remove this line from here.

rav-indra replied:
done

ravin-essence commented on app/views/dashboard/_buyer.html.erb:
```diff
@@ -1,15 +1,45 @@
-<h1 style="margin-top: 120px; text-align:center;">Purchased Properties</h1>
 <% props = Property.where(buyer_id: current_user.id) %>
 
+<h1 style="margin-top: 130px; text-align:center;">Purchased Properties</h1>
```
Per page only 1 H1 is allowed.

rav-indra replied:
- Done

ravin-essence commented on app/views/dashboard/_buyer.html.erb:
```diff
@@ -1,15 +1,45 @@
+  <div class="container">
+    <div class="row">
+      <% props.each do |property| %>
+        <% seller = User.find(property.user_id) %>
```
Remove this line and move on to the controller.

rav-indra replied:
- Moved lines to controller

rav-indra replied:
- Done

ravin-essence commented on app/views/dashboard/_buyer.html.erb:
```diff
@@ -1,15 +1,45 @@
+            </div>
+            <div class="col-sm-12 col-md-6" style="float:left;">
+              <div class="testimonial-author-box text-center">
+                <h2>Property Details:</h2>
```
manage these PROPERTY-DETAILS and SELLER-DETAILS into a separate page.

rav-indra replied:
- Made a partial for this

ravin-essence commented on app/views/dashboard/_buyer.html.erb:
```diff
@@ -1,15 +1,45 @@
+              <div class="content">
+                <p class="testimonial-text fs-5">
+                  <b>Name: </b><%=  property.name %><br/>
+                  <b>Description: </b><%=  property.description %><br/>
```
Remove extra spacing from here.

rav-indra replied:
- Removed extra spaces

ravin-essence commented on app/views/dashboard/_buyer.html.erb:
```diff
@@ -1,15 +1,45 @@
+              </div>
+              <div class="content">
+                <p class="testimonial-text fs-5">
+                  <b>Name: </b><%=  seller.name %><br/><br/><br/>
```
Manage this 3 br using css styling.

rav-indra replied:
- Removed

ravin-essence commented on app/views/dashboard/_seller.html.erb:
```diff
@@ -1,9 +1,6 @@
-<% sold.each do |sld| %>
-  <% sum = sum + sld.price %>
-<% end %>
+<% sold_property =  props.where(is_paid: true) %>
```
same as above.

rav-indra replied:
- Moved this code to controller

ravin-essence commented on app/views/dashboard/_seller.html.erb:
```diff
@@ -1,9 +1,6 @@
-<% end %>
+<% sold_property =  props.where(is_paid: true) %>
+
 <div class="row" style="margin-top: 130px; justify-content: center;">
```
Remove inline styling.

rav-indra replied:
- Removed inline styles and added it to CSS

ravin-essence commented on app/views/dashboard/sold.html.erb:
```diff
@@ -0,0 +1,79 @@
+    <div class="row">
+      <% sold_property.each do |property| %>
+        <% buyer = User.find(property.buyer_id) %>
+        <div class="box">
```
use separate partial for this content.

rav-indra replied:
- Made partial

ravin-essence commented on app/views/dashboard/sold.html.erb:
```diff
@@ -0,0 +1,79 @@
+  </div>
+</section>
+
+
```
remove extra spacing.

rav-indra replied:
- Removed extra spaces

ravin-essence commented on app/views/dashboard/sold.html.erb:
```diff
@@ -0,0 +1,79 @@
+                <span class="color-text-a">Phone .</span> +91 12345 67890
+              </li>
+              <li class="color-a">
+                <span class="color-text-a">Email .</span> <%= mail_to "contact@propex.com", "contact@propex.com", target: '_blank' %>
```
Can we make it like...
 
 <span class="color-text-a">Email .</span>
  <%= mail_to "contact@propex.com", "contact@propex.com", target: '_blank' %>

rav-indra replied:
- Done

ravin-essence commented on app/views/devise/registrations/edit.html.erb:
```diff
@@ -1,4 +1,4 @@
-<div class="text-center"><h2>Edit <%= resource_name.to_s.humanize %></h2>
+<br/><div class="text-center"><h2>Edit <%= resource_name.to_s.humanize %></h2>
```
Please try to manage using css.

rav-indra replied:
- Done

ravin-essence commented on app/views/layouts/nav/_public.html.erb:
```diff
@@ -6,7 +6,7 @@
         <span></span>
         <span></span>
       </button>
-      <a class="navbar-brand text-brand" href="/"><h1><b>Prop<span class="color-b">Ex</span></b></h1></a>
```
Remove h1

rav-indra replied:
- Removed

ravin-essence commented on app/views/properties/index.html.erb:
```diff
@@ -29,7 +29,7 @@
-</div>
+</div><br/>
     <!-- ======= Property Grid ======= -->
     <section class="property-grid grid">
```
correct the indentation.

rav-indra replied:
- Corrected all identetions

ravin-essence commented on app/views/properties/new.html.erb:
```diff
@@ -5,6 +5,39 @@
+
+<%# ---- Footer ---- %>
+<section class="section-footer">
+  <div class="container">
```
move this footer as a common partial, So we can **reuse** it.

rav-indra replied:
- Made a partial

rav-indra committed:
Change/identation-css-space

rav-indra committed:
Shortend the code of defferntiate user

rav-indra committed:
Made few changes

rav-indra commented:
- Removed extra spaces
- Removed extra h1 tags from pages
- Fixed Indentation
- Made partials where it was needed
- Moved inline style to CSS file
- Moved code from view to controller
- Made helpers to reduce repeated code and reuse it

rav-indra committed:
Changed form error,code of dashboard

rav-indra commented:
- Commit

rav-indra commented:
- All done

rav-indra commented on app/helpers/application_helper.rb:
```diff
@@ -26,6 +26,14 @@ def profile(role)
     end
   end
 
+  def c_user(value)
```
- Done

rav-indra commented on app/views/dashboard/_seller.html.erb:
```diff
@@ -1,50 +1,47 @@
-                  <div class="col mr-2">
-                      <div class="text-xs font-weight-bold text-success text-uppercase mb-1 fs-5">
-                          My Properties</div>
-                      <div class="h5 mb-0 font-weight-bold text-gray-800" style="margin-top:-30px; text-align:end; font-size:xxx-large;"><%= props.count %></div>
```
- Done

rav-indra commented on app/views/dashboard/_seller.html.erb:
```diff
@@ -1,50 +1,47 @@
+          <div class="col mr-2">
+            <div class="text-xs font-weight-bold text-success text-uppercase mb-1 fs-5">
+              My Properties</div>
+            <div class="detail_box h5 mb-0 font-weight-bold text-black"><%= @props.count %></div>
```
- Done

rav-indra commented on app/views/dashboard/index.html.erb:
```diff
@@ -1,38 +1,5 @@
-<% end %>
-
+<div class="dashboard">
+  <%= render current_user.role %>
```
- Didn't need

rav-indra commented on app/views/dashboard/_admin.html.erb:
```diff
@@ -19,6 +18,6 @@
         <%#= link_to "delete", user_registration_path(user), :method => :delete, :confirm => "You sure?" %>
         <%#= button_to "Delete", method: :delete, class: "btn btn-outline-danger" %>
```
- Done

rav-indra commented on app/views/dashboard/_admin.html.erb:
```diff
@@ -1,16 +1,15 @@
-      <% users.each do |user| %>
+    <% @users.each do |user| %>
       <tr>
         <th scope="row"><%= link_to image_tag(profile(user.role))%></th>
```
- Done

rav-indra commented on app/helpers/application_helper.rb:
```diff
@@ -16,6 +15,7 @@ def resource_class
     @devise_mapping ||= Devise.mappings[:user]
   end
+
   def profile(role)
```
- Done

rav-indra commented:
- All done
